stable_baselines3/common/distributions.py

```diff
@@ -28,7 +38,7 @@ def proba_distribution_net(self, *args, **kwargs) -> Union[nn.Module, Tuple[nn.M
         concrete classes."""
 
     @abstractmethod
-    def proba_distribution(self, *args, **kwargs) -> "Distribution":
+    def proba_distribution(self: SelfDistribution, *args, **kwargs) -> SelfDistribution:
```

Comment:
is the `self: SelfDistribution` needed?

looks like duplicate info.

Reply:
It is. And yes I seems duplicate, that's the [motivation behind the new Self annotation](https://peps.python.org/pep-0673/) that we do not use for the moment (see https://github.com/DLR-RM/stable-baselines3/pull/1043#issuecomment-1240437433):

> The present workaround for such instances is to define a `TypeVar` with the base class as the bound and use it as the annotation for the `self` parameter and the return type:
> 
> ```python
> from typing import TypeVar
> 
> TShape = TypeVar("TShape", bound="Shape")
> 
> class Shape:
>     def set_scale(self: TShape, scale: float) -> TShape:
>         self.scale = scale
>         return self
> 
> 
> class Circle(Shape):
>     def set_radius(self, radius: float) -> Circle:
>         self.radius = radius
>         return self
> 
> Circle().set_scale(0.5).set_radius(2.7)  # => Circle
> ```
> 
> Unfortunately, this is verbose and unintuitive. Because `self` is usually not explicitly annotated, the above solution doesn’t immediately come to mind, and even if it does, it is very easy to go wrong by forgetting either the bound on the `TypeVar(bound="Shape")` or the annotation for self.